### ISSUE TITLE
generated webapp uses a pretty old version of the postgresql JDBC driver

### ISSUE THREAD

**Issue by pmauduit** [georchestra/mapstore2-georchestra#431]
After having built the webapp war, we have this following postgresql JDBC driver in the classpath:

```
postgresql-8.4-702.jdbc3.jar
```

This is a pretty outdated version which is not compatible with only-SSL enabled servers (the `sslmode=require` parameter). Would it be possible to consider upgrading it ?

I could not find where the (maven) dependency was declared, so I cannot suggest a PR here. But testing with the latest version of the driver fixed our issue with SSL-only postgresql servers.

**Comment by Gaetanbrl** [comment]
Maybe to link with this issue ?
https://github.com/geosolutions-it/MapStore2/issues/7272

**Comment by Gaetanbrl** [comment]
This pg driver version seems changed in https://github.com/geosolutions-it/geostore/pull/238/files#diff-384baf9d9dd5407c10719d4da53cf994cd22f7da46e57405cfd9249f53d213a9

**Comment by pmauduit** [comment]
> So, i suppose that the next upgrade between mapstore2 > mapstore2-georchestre will upgrade this pg version.

Thanks, it indeed seems related to this similar mentioned issue. is there an ETA planned for this upgrade ?

**Comment by Gaetanbrl** [comment]
No idea... @catmorales ?

**Comment by catmorales** [comment]
I don't know .
This has not been mentioned

**Comment by offtherailz** [comment]
Correct. The [issue](https://github.com/geosolutions-it/MapStore2/issues/7272) and the commit you mentioned are on MapStore master branch, therefore the update will be available when geOrchstra will update to 2022.01.xx (master).

**Comment by landryb** [comment]
looking at the issue, i'm not sure if the incompatibility is with psql > 12 (which would be a major annoyance, im using 12 so far but planning to update to 14 soon) or 'just' with SSL-only servers and/or SCRAM-SHA-256 passwords (which is still an issue, less 'major' since we can always fallback to MD5 passwords for mapstore). Either way, can this be clarified ?

Good thing that one can directly replace the jar within the war by the latest version to workaround the issue :)

It also seems the PR is candidate for backport to 2021.03.xx, correct ? If so, maybe it can also be backported to 2021.02.xx ?

Also, i dunno if the jar matches the psql version, but that postgresql jar is from 2010 according to https://mvnrepository.com/artifact/postgresql/postgresql and for jdk olders than java6... and its archived upstream at https://jdbc.postgresql.org/download.html, so pretty old by my standards :)

**Comment by pmauduit** [comment]
> Either way, can this be clarified ?

What motivated opening the issue here from my side: I could not have mapstore2-georchestra successfully starting, because we are using a postgresql instance which does not accept non-SSL connections, even after having hard-patched the postgresql JDBC driver & adding the `sslmode` parameter in the url.

Removing the postgres jdbc driver (see mentioned jar) and upgrading to the latest version available relying on mvnrepository.com directly into the ms2 classpath made it.

**Comment by landryb** [comment]
i've just realized that my ms2/geor testing instance runs psql server 13.5 with md5 passwords and without ssl, no issues, so the incompatibility is really with SSL-only servers and/or SCRAM-SHA-256 - either way, would be nice to have it fixed properly instead of having to swap psql jars under the hood :)

**Comment by offtherailz** [comment]
> looking at the issue, i'm not sure if the incompatibility is with psql > 12 (which would be a major annoyance, im using 12 so far but planning to update to 14 soon) or 'just' with SSL-only servers and/or SCRAM-SHA-256 passwords (which is still an issue, less 'major' since we can always fallback to MD5 passwords for mapstore). Either way, can this be clarified ?
> 
> Good thing that one can directly replace the jar within the war by the latest version to workaround the issue :)
> 
> It also seems the PR is candidate for backport to 2021.03.xx, correct ? If so, maybe it can also be backported to 2021.02.xx ?
> 
No at the moment is not candidate for backport. I will ask if it can be backported without issues. 

> Also, i dunno if the jar matches the psql version, but that postgresql jar is from 2010 according to https://mvnrepository.com/artifact/postgresql/postgresql and for jdk olders than java6... and its archived upstream at https://jdbc.postgresql.org/download.html, so pretty old by my standards :)

**Comment by tdipisa** [comment]
The lib update is present on MS master at the moment and it is scheduled to be released as part of the next major version (2022.01.00). For some other client projects (where postgres13 is in use) we have anyway used/tested the newer postgresql dependency at MS project level without problems.
